### PR TITLE
Poco: Proper options to handle ActiveRecord and ActiveRecordCompiler

### PR DESCRIPTION
**Pull request by lbakman**
Library: Poco
Tested with versions: 1.10.1, 1.11.1 and 1.11.2

Fixes: #8266.

I have been using a custom recipe for some time, because of the problem with the ActiveRecord option. This is my attempt at fixing the issue, so I can use the proper recipe and others do not have the same problem that I do.

Renamed option "enable_active_record" to "enable_activerecord" to match the option in the CMakeLists.txt file.
- The "enable_active_record" option never worked. This will allow the ActiveRecord component to be disabled from the recipe.

Added option "enable_activerecord_compiler" to allow the ActiveRecordCompiler to be built using the recipe.
- Like other tools, it is now possible to 'opt in' to build the ActiveRecord compiler tool.
- Not adding Foundation as a dependency. Other non-library components do not specify Foundation either.
- Adding XML as an explicit dependency, even though the CMakeLists.txt does not. There is an implicit dependency to XML, and by adding it to the recipe, it will fail fast when being executed without the right options.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by lbakman** [comment]
@obiltschnig, may I request a review of this pull-request? Perhaps as part of the update to version 1.11.3?

**Review by obiltschnig — DISMISSED** [review]

File: recipes/poco/all/conanfile.py [hunk 1]
```diff
@@ -102,9 +105,12 @@ def config_options(self):
             del self.options.enable_data_postgresql
             del self.options.enable_jwt
         if tools.Version(self.version) < "1.11":
-            del self.options.enable_active_record
+            del self.options.enable_activerecord
+            del self.options.enable_activerecord_compiler
 
     def configure(self):
+        if self.options.enable_active_record != "deprecated":
+            self.output.warn("enable_active_record option is deprecated, do not use anymore.")
```
Comment (uilianries) [inline recipes/poco/all/conanfile.py:113]:
```suggestion
            self.output.warn("enable_active_record option is deprecated, use 'enable_activerecord' instead")
```

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/poco/all/conanfile.py [hunk 2]
```diff
@@ -126,7 +132,7 @@ def requirements(self):
             self.requires("apr-util/1.6.1")
         if self.options.enable_netssl or self.options.enable_crypto or \
            self.options.get_safe("enable_jwt"):
-            self.requires("openssl/1.1.1n")
+            self.requires("openssl/1.1.1o")
```
Comment (tuduongquyet) [inline recipes/poco/all/conanfile.py:135]: Should we upgrade sqlite3 too?

Comment (lbakman) [reply recipes/poco/all/conanfile.py:135]: I would rather not mix other upgrades into this pull-request. 
The update of openssl was mandated by the build system to actually build. An update of sqlite is not. 
If you need a newer version of sqlite in your project, you can always override the version in your project.

Comment (uilianries) [reply recipes/poco/all/conanfile.py:135]: please, update to the latest sqlite version too. It avoids other warnings when consuming Poco in Conan center and sqlite is outdated. There is no big deal updating dependencies in same PR, it's an usual action.

Comment (lbakman) [reply recipes/poco/all/conanfile.py:135]: I have updated sqlite and libmysqlclient as well. Updating libmysqlclient resolves a warning about different versions of openssl.

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by lbakman** [comment]
@obiltschnig @tuduongquyet May I re-request a review? For some reason the "Re-request review" buttons does not work properly for me.

**Review by SSE4 — APPROVED** [review]

**Pull request merged**